storage/innobase/include/trx0trx.h

```diff
@@ -1322,6 +1450,100 @@ trx_is_started(
 	       && trx->state != TRX_STATE_FORCED_ROLLBACK);
 }
 
+inline ib_uint64_t
+trx_stats::start_io_read(const trx_t& trx, ulint bytes)
+{
+	if (bytes) {
+		thd_report_innodb_stat(trx.mysql_thd, trx.id,
+				       MYSQL_TRX_STAT_IO_READ_BYTES, bytes);
+	}
+	ulint sec, ms;
+	ut_usectime(&sec, &ms);
+	return static_cast<ib_uint64_t>(sec * 1000000 + ms);
+}
+
+inline ib_uint64_t
+trx_stats::start_io_read(trx_t *trx, ulint bytes)
+{
+	if (UNIV_LIKELY_NULL(trx))
+		return trx->stats.start_io_read(*trx, bytes);
```

Comment:
Why not just `return start_io_read(*trx, bytes);`?

Reply:
Here and everywhere below: the fast-path through extra logging is with InnoDB extensions disabled, and that happens when trx == NULL

Reply:
Still not quiet get it. My main question why do we call `STATIC` methods (like `start_io_read()`) via a pointer to object

Reply:
Sorry I misunderstood, I thought you suggested removing the if () statement. Yes, will not call static methods through object pointer